README.md

```diff
@@ -49,41 +49,56 @@ $ sha256sum -c --strict SHA256SUM.md
-  -print-provenance
-    	output the verified provenance
-  -provenance string
-    	path to a provenance file
-  -source string
-    	expected source repository that should have produced the binary, e.g. github.com/some/repo
-  -tag string
-    	[optional] expected tag the binary was compiled from
-  -versioned-tag string
-    	[optional] expected version the binary was compiled from. Uses semantic version to match the tag
+$ go run ./cli/slsa-verifier/ verify-artifact --help
+Verifies SLSA provenance on an artifact blob
+
+Usage:
+  slsa-verifier verify-artifact [flags]
+
+Flags:
+      --branch string            [optional] expected branch the binary was compiled from
+      --builder-id string        EXPERIMENTAL: the unique builder ID who created the provenance
+  -h, --help                     help for verify-artifact
+      --print-provenance         print the verified provenance to std out
+      --provenance-path string   path to a provenance file
+      --source string            expected source repository that should have produced the binary, e.g. github.com/some/repo
+      --tag string               [optional] expected tag the binary was compiled from
+      --versioned-tag string     [optional] expected version the binary was compiled from. Uses semantic version to match the tag
+      --workflow-input map[]     [optional] a workflow input provided by a user at trigger time in the format 'key=value'. (Only for 'workflow_dispatch' events). (default map[])
 ```
 
 ### Example
 
 ```bash
-$ go run ./cli/slsa-verifier -artifact-path ~/Downloads/slsa-verifier-linux-amd64 -provenance ~/Downloads/slsa-verifier-linux-amd64.intoto.jsonl -source github.com/slsa-framework/slsa-verifier -tag v1.3.0
+$ go run ./cli/slsa-verifier -provenance-path ~/Downloads/slsa-verifier-linux-amd64.intoto.jsonl --source github.com/slsa-framework/slsa-verifier --tag v1.3.0 ~/Downloads/slsa-verifier-linux-amd64 
 Verified signature against tlog entry index 3189970 at URL: https://rekor.sigstore.dev/api/v1/log/entries/206071d5ca7a2346e4db4dcb19a648c7f13b4957e655f4382b735894059bd199
 Verified build using builder https://github.com/slsa-framework/slsa-github-generator/.github/workflows/builder_go_slsa3.yml@refs/tags/v1.2.0 at commit 5bb13ef508b2b8ded49f9264d7712f1316830d10
 PASSED: Verified SLSA provenance
 ```
 
 The verified in-toto statement may be written to stdout with the `--print-provenance` flag to pipe into policy engines.
 
+### Options Details
+
+For provenance created using SLSA GitHub generators:
+
+| Option | Description |
+| --- | ----------- |
+| `source` | Expects a source, for e.g. `github.com/org/repo`. This option verifies that `ConfigSource` and `Materials` of the SLSA provenance contain the expected `source`. |
+| `branch` | Expects a `branch` like `main` or `dev`. Verifies this against the `github_ref` in the SLSA provenance when the `github_ref_type` is a `branch`, or the `base_ref` or `target_commitish` when a `tag`.|
```

Comment:
I wonder if we need this level of details here. I'm worried it will go out of sync with the implementation.
How about:
1. Keep the documentation simpler in this README. Maybe we can have a table that explains which options are supported for which builder, and that the branch option is brittle.
2. If we want more detailed doc, maybe have a README under `verifiers/internal/<builder>/README.md`. There is still a risk of getting out of sync, of course....

Anything we could do better that what I'm proposing?

Reply:
Simplified the README.md. I mostly want to make sure that there's documentation for what the options should look like, with some examples in this PR

Eveneually we should have some documentation somewhere of how verification works for specific builders.